src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs

```diff
@@ -61,6 +61,8 @@ public bool IsIndeterminate
 
         [Parameter] public EventCallback<bool> OnChange { get; set; }
 
+        private bool hasSetIsChecked { get; set; }
```

Comment:
It is not a good name for this property, I think it should be named something like "IsCheckedHasValue" or "IsCheckHasBeenSet"

Reply:
And change this property into a private field

Reply:
Changed hasSetIsChecked  to isCheckedHasValue

Reply:
after some investigation, I think the "IsCheckedHasBeenSet" is a better name. please apply this particular pattern for naming this specific field, which will be applied in the source generator in near future.

Reply:
Rename hasSetIsChecked to IsCheckedHasBeenSet